src/plugins/api-proxy/util/receive.js

```diff
@@ -0,0 +1,250 @@
+import querystring from 'qs';
+
+import { API_PROXY_PLUGIN_NAME } from '../';
+import { coerceBool, toCamelCase } from './stringUtils';
+
+// match escpaed unicode characters that are treated as newline literals in JS
+const ESCAPED_UNICODE_NEWLINES = /\\u2028|\\u2029/g;
+
+/**
+ * Convert the X-Meetup-Variants response header into a state-ready object
+ *
+ * @see {@link https://meetup.atlassian.net/wiki/display/MUP/X-Meetup-Variants}
+ * @returns {Object} {
+ *   [experiment]: {
+ *     [context (member/chapter id)]: variantName
+ *   }
+ * }
+ */
+export const parseVariantsHeader = variantsHeader =>
+	variantsHeader.split(' ').reduce((variants, keyval) => {
+		const [experiment, val] = keyval.split('=');
+		variants[experiment] = variants[experiment] || {};
+		const [context, variant] = val.split('|');
+		variants[experiment][context] = variant || null;
+		return variants;
+	}, {});
+
+const X_HEADERS = ['x-total-count'];
+
+export const parseMetaHeaders = headers => {
+	const meetupHeaders = Object.keys(headers)
+		.filter(h => h.startsWith('x-meetup-'))
+		.reduce((meta, h) => {
+			const key = toCamelCase(h.replace('x-meetup-', ''));
+			meta[key] = headers[h];
+			return meta;
+		}, {});
+
+	// special case handling for flags
+	if (meetupHeaders.flags) {
+		meetupHeaders.flags = querystring.parse(meetupHeaders.flags, {
+			delimiter: ', ',
+			decoder: coerceBool,
+		});
+	}
+
+	// special case handling for variants
+	if (meetupHeaders.variants) {
+		meetupHeaders.variants = parseVariantsHeader(meetupHeaders.variants);
+	}
+
+	const xHeaders = X_HEADERS.reduce((meta, h) => {
+		const key = toCamelCase(h.replace('x-', ''));
+		if (h in headers) {
+			meta[key] = headers[h];
+		}
+		return meta;
+	}, {});
+
+	const linkHeader =
+		headers.link &&
+		headers.link.split(',').reduce((links, link) => {
+			const [urlString, relString] = link.split(';');
+			const url = urlString.replace(/<|>/g, '').trim();
+			var rel = relString.replace(/rel="([^"]+)"/, '$1').trim();
+			links[rel] = url;
+			return links;
+		}, {});
+	const meta = {
+		...meetupHeaders,
+		...xHeaders,
+	};
+
+	if (linkHeader) {
+		meta.link = linkHeader;
+	}
+	return meta;
+};
+
+/**
+ * Accept an Error and return an object that will be used in place of the
+ * expected API return value
+ * @param {Error} err the error to populate
+ * @return {Object} { value, error? }
+ */
+function formatApiError(err) {
+	return {
+		value: null,
+		error: err.message,
+	};
+}
+
+/**
+ * @return {Object} { value, error? }
+ */
+export const parseApiValue = ([response, body]) => {
+	// treat non-success HTTP code as an error
+	if (response.statusCode < 200 || response.statusCode > 299) {
+		return formatApiError(new Error(response.statusMessage));
```

Comment:
happy to change, although I have no idea how the api proxy would handle a redirect response - make another request to the redirected endpoint? I don't think it would actually return a value, but I guess it's fine to let a 3xx return a null `value` for now instead of erroring. (I actually need to merge #336 in order to get some of this behavior smoothed out).